test/libsolidity/syntaxTests/operators/custom/implicit_conversion_is_blocked.sol

```diff
@@ -0,0 +1,34 @@
+using {add as +, unsub as -} for S;
+
+struct S {
+    uint x;
+}
+
+function add(S memory _a, S memory) returns (S memory) {
+    return _a;
+}
+
+function unsub(S memory _a) returns (S memory) {
+    return _a;
+}
+
+contract C {
+    S s;
+
+    function test() public {
+        S memory sTmp;
+        s + s;
+        sTmp + true;
+        true + s;
+        -sTmp;
+        -s;
+        -true;
+    }
+}
+
+// ----
+// TypeError 2271: (288-293): Operator + not compatible with types struct S storage ref and struct S storage ref. No matching user-defined operator found.
+// TypeError 5653: (303-314): User defined operator + not compatible with types struct S memory and bool.
+// TypeError 2271: (324-332): Operator + not compatible with types bool and struct S storage ref.
```

Comment:
Maybe it would make sense to add to the error message what kind of operator was not compatible. E.g. ` Binary operator +` not compatible or `Unary operator +`. Because in the worst-case the developer need to "search" which operator was exactly creating the error, e.g. unary/binary. I think this information would be nice to have in general, whenever an operator is mentioned in an error/warning message.